crates/evm/evm.rs

```diff
@@ -486,3 +487,23 @@ fn access_list_inspector(
         precompile_addresses,
     ))
 }
+
+/// Returns the spec id according to the block timestamp and the stored chain config
+/// WARNING: Assumes at least Merge fork is active
+pub fn spec_id(store: &Store, block_timestamp: u64) -> Result<SpecId, StoreError> {
```

Comment:
imo this calculation should be done outside the evm and passed as spec_id. I don't think it's a good idea to couple the evm crate to our store implementation.

Thinking more broadly we should have a `import_block`/`process_block` that is in core, calls this function and makes updates the state.

If that's the case, would it be possible to remove the storage crate dependency from this crate?

Thoughts?

Reply:
We cannot import neither the evm crate nor the storage crate from core due to cyclic dependencies. We cannot remove the storage dependency from the evm crate either as the evm reads from the `Store` during execution

Reply:
We could move it into an `EvmState` method so it doesn't appear as disconnected?

Reply:
Hmm, ok, this needs some more thought. Let's merge this one as is and we can think some further refactor down the line.